### PR TITLE
Add conversions for Value to/from C++ structs

### PR DESCRIPTION
**Pull request by emaxx-google**
Implement helpers for semi-automated implementation that maps between
C++ structs and JSON-like dictionary Values.

This extends the set of helpers for converting between Value instances
and C/C++ objects, paving the way for migrating the rest of //common/cpp
library to become toolchain-independent (as tracked by #185).

This commit implements analogous functionality similar to the already
existing Native Client pp::Var specific helpers (in
pp_var_utils/struct_converter.h), however this new implementation requires
much less boilerplate code to be written for every struct.

### CODE REVIEW

**Review by Fabian-Sommer — APPROVED** [review]
LGTM % nits

File: common/cpp/src/google_smart_card_common/value_conversion.h [hunk 1]
```diff
@@ -174,7 +307,92 @@ class EnumValueDescriptor {
...
+          from_value_converter_(from_value_converter) {}
+    Description(Description&&) = default;
+    Description& operator=(Description&&) = default;
+    ~Description() = default;
+
+    const char* type_name() const { return type_name_; }
+
+    // Adds the given field into the struct's description: |dictionary_key_name|
+    // is the key in the dictionary `Value` representation of |field_ptr|.
+    // Returns a rvalue reference to |this| and uses the "&&" ref-qualifier, so
```
Comment (MahmoudAGawad) [inline common/cpp/src/google_smart_card_common/value_conversion.h:351]: nit: Returns **an** rvalue

**Review by MahmoudAGawad — APPROVED** [review]
LGTM.

File: common/cpp/src/google_smart_card_common/value_conversion.h (hunk 1 above)
Comment (emaxx-google) [reply common/cpp/src/google_smart_card_common/value_conversion.h:351]: Done. Apologies, missed this after copy-pasting the unreviewed comment.

**Comment by emaxx-google** [comment]
Thanks! That was the last "monster" CL hopefully :)

**Pull request merged**